code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -239,12 +267,14 @@ While using this makes the system rely on OnFire, it still gives options for tim
 	var/turf/T = get_turf(src)
 	if(loc == null)
 		return
+	var/activator_ref = REF(activator)
+	var/mychild_ref = REF(mychild)
 	for(var/t in RANGE_TURFS(12, T))
 		if(get_dist(t, T) == 12)
 			var/obj/effect/temp_visual/elite_tumor_wall/newwall
 			newwall = new /obj/effect/temp_visual/elite_tumor_wall(t, src)
-			newwall.activator = src.activator
-			newwall.ourelite = src.mychild
+			newwall.activator_ref = activator_ref
+			newwall.ourelite_ref = mychild_ref
```

Comment:
Why not use weakrefs for activator? it'd make this process a bit smoother

Reply:
To be honest with you I meant as opposed to signals, since they can break in fun and interesting ways if someone ever tries the same solution somewhere else in the proc path, but I don't have a good idea of the scope of the activator var
Is it used a lot in different procs?

Reply:
The activator (which is basically the challenger) and mychild (which is the elite mining mob) variables are used in about every `/obj/structure/elite_tumor` proc, for stuff that ranges from victory/loss conditions, heirophant-like barriers that stop one or another from leaving the 24x24 arena box, loot etcetera. Adding weakref.resolve() calls to each one of them would be too tedious.
A qdeleting signal proc should be safe enough already given how the same solution should be limited to endemic/intrinsic variable outside of components, elements and in general datums with a low tree structure height.

Reply:
I bring it up because a good few of those uses are me, and it's bit me in the ass before. If there's a fuck ton of uses it's fine, but I'd prefer not to overuse the tool because of the headaches it can cause.

Reply:
I'll take note of that. Perhaps @IndieanaJones might also do something about it if he decides to rework tumours one day.